utils/src/main/scala/za/co/absa/enceladus/utils/fs/FileReader.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.fs
+
+import scala.io.Source
+
+trait FileReader {
+  def readFileAsLines(filename: String): Seq[String] = {
```

Comment:
Hehe, we discussed this with @yruslan on Friday.
`Vector` is an oddly specific type to return from a function such this. `Seq` feels much more natural. But it was necessary to copy the file content to a new structure, and `vector` is generally thought as the more performant (than `list` for example) .

Reply:
Vector is faster on accessing elements by index. When processing elements sequentially starting from the beginning, `List` and `Vector` have the same performance profile.
From my personal perspective, I slightly prefer `List` since it is more lightweight. But I'm fine with `Vector` here as well.

As for return type, it might be something to discuss on CQC. We might come up with the rules when to return an interface (e.g. `Seq`), and when to return a concrete type.

Reply:
[Vector performance](https://docs.scala-lang.org/overviews/collections/performance-characteristics.html) by scala-lang.org.

I would go for `List` as well, we don't gain much by using `Vector` and I think `List` is expected from `getLines` in general. So keeping it together with scala.

Reply:
List it is then. I can even rename the function and then go with `List` as return type...